pymongo/srv_resolver.py

```diff
@@ -56,7 +56,7 @@ def _resolve(*args: Any, **kwargs: Any) -> resolver.Answer:
 class _SrvResolver:
     def __init__(
         self,
-        fqdn: str,
+        fqdn: Optional[str],
```

Comment:
Is fqdn really optional?

Reply:
I thought it was based on line 335 of monitor.py which passes `SrvMonitor._fqdn` into _SrvResolver as `fqdn` and `SrvMonitor._fqdn` is defined by taking `TopologySettings.fqdn` which is an `Optional[str]`

Reply:
It's optional on TopologySettings but it's required for SrvMonitor and _SrvResolver.

Reply:
I see, just fixed it!